engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompilerData.scala

```diff
@@ -70,8 +78,15 @@ class FlinkProcessCompilerData(
   def restartStrategy: RestartStrategies.RestartStrategyConfiguration = exceptionHandler.restartStrategy
 
   def prepareExceptionHandler(runtimeContext: RuntimeContext): FlinkExceptionHandler = {
-    exceptionHandler.open(FlinkEngineRuntimeContextImpl(jobData, runtimeContext))
-    exceptionHandler
+    componentUseCase match {
+      case ComponentUseCase.TestRuntime =>
+        exceptionHandler.open(FlinkTestEngineRuntimeContextImpl(jobData, runtimeContext))
```

Comment:
DRY. after removing duplication in open, you'll get a nice method that can be used here.